`spack install` output is being inserted into autoconf generated `configure` script

I'm trying to add a newer release of the kallisto package, ideally 0.46.2.

Kallisto is a CMakePackage and/but is "interesting" (may you live in interesting times) because it has vendored its own copy of the `htslib` library and runs `autoheader` and `autoconf` and `configure` in the `htslib` dir via this bit of the toplevel `CMakeLists.txt`:

```
include(ExternalProject)
ExternalProject_Add(htslib
    PREFIX ${PROJECT_SOURCE_DIR}/ext/htslib
    SOURCE_DIR ${PROJECT_SOURCE_DIR}/ext/htslib
    BUILD_IN_SOURCE 1
    CONFIGURE_COMMAND autoheader && autoconf && ${PROJECT_SOURCE_DIR}/ext/htslib/configure
        --prefix=${PREFIX} --disable-bz2 --disable-lzma --disable-libcurl
    BUILD_COMMAND make lib-static
    INSTALL_COMMAND ""
)
```

Anything newer than the version we have in the tree (0.43.1) dies with a similar problem, it seems like the output of the `spack` command is being included in the generated files.

Here's the first few lines of the generated `configure` script

```
#! /bin/sh
# Guess values for system-dependent variables and create Makefiles.
# Generated by GNU Autoconf 2.69 for HTSlib make3: Entering directory `/tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib'
1.4.1
make3: Leaving directory `/tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib'.
#
# Report bugs to <samtools-help@lists.sourceforge.net>.
#
```

The build dies when cmake tries to build the vendored `htslib` lib and stubs its toe on the line tht only contains `1.4.1`:

```
[...]
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

4 errors found in build log:
     109    configure.ac:26: 1.4.1
     110    configure.ac:26: make3: Leaving directory `/tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib'
     111    configure.ac:26: warning: AC_INIT: not a literal: make3: Entering directory `/tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spac
            k-src/ext/htslib'
     112    configure.ac:26: 1.4.1
     113    configure.ac:26: make3: Leaving directory `/tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib'
     114    /tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib/configure: line 4: 1.4.1: command not found
  >> 115    /tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib/configure: line 34: syntax error near unexpected token `('
     116    /tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib/configure: line 34: `  case `(set -o) 2>/dev/null` in #('
  >> 117    make[2]: *** [/tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-src/ext/htslib/src/htslib-stamp/htslib-configure] Error 2
     118    make[2]: Leaving directory `/tmp/ghartzell/spack-stage/spack-stage-kallisto-0.46.2-2cya5t4j2epry6ttce7w2bm5eeoqurjf/spack-build'
  >> 119    make[1]: *** [CMakeFiles/htslib.dir/all] Error 2
     120    make[1]: *** Waiting for unfinished jobs....
[...]
```

On the theory that I was picking up old CentOS versions of the tools, I tried (unconditionally, for now) adding dependencies on the usual autotools suspects, to no avail.  Here's the changes:

```
$ git diff
diff --git a/var/spack/repos/builtin/packages/kallisto/package.py b/var/spack/repos/builtin/packages/kallisto/package.py
index 337b50f18..81d509cb4 100644
--- a/var/spack/repos/builtin/packages/kallisto/package.py
+++ b/var/spack/repos/builtin/packages/kallisto/package.py
@@ -13,8 +13,14 @@ class Kallisto(CMakePackage):
     homepage = "http://pachterlab.github.io/kallisto"
     url      = "https://github.com/pachterlab/kallisto/archive/v0.43.1.tar.gz"

+    version('0.46.2', sha256='c447ca8ddc40fcbd7d877d7c868bc8b72807aa8823a8a8d659e19bdd515baaf2')
     version('0.43.1', sha256='7baef1b3b67bcf81dc7c604db2ef30f5520b48d532bf28ec26331cb60ce69400')

     depends_on('zlib')
     depends_on('hdf5')
     depends_on('mpich')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
$
```

--- a/var/spack/repos/builtin/packages/kallisto/package.py
+++ b/var/spack/repos/builtin/packages/kallisto/package.py
@@ -20,12 +20,15 @@ class Kallisto(CMakePackage):
     depends_on('hdf5')
     depends_on('mpich')
 
-    depends_on('autoconf', type='build')
-    depends_on('automake', type='build')
-    depends_on('libtool',  type='build')
-    depends_on('m4',       type='build')
     # htslib isn't built in time to be used....
     parallel = False
+
+    # v0.44.0 vendored a copy of htslib and uses auto* to build
+    # its configure script.
+    depends_on('autoconf', type='build', when='@0.44.0:')
+    depends_on('automake', type='build', when='@0.44.0:')
+    depends_on('libtool',  type='build', when='@0.44.0:')
+    depends_on('m4',       type='build', when='@0.44.0:')
 
     # Including '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' in the cmake args
     # causes bits of cmake's output to end up in the autoconf-generated
@@ -38,6 +41,9 @@ class Kallisto(CMakePackage):
         setting.
         """
         a = super(Kallisto, self).std_cmake_args
-        args = [i for i in a if i != '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON']
+        if (self.spec.version >= Version('0.44.0')):
+            args = [i for i in a if i != '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON']
+        else:
+            args = a
 
         return args

--- a/var/spack/repos/builtin/packages/kallisto/package.py
+++ b/var/spack/repos/builtin/packages/kallisto/package.py
@@ -24,6 +24,8 @@ class Kallisto(CMakePackage):
     depends_on('automake', type='build')
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
+    # htslib isn't built in time to be used....
+    parallel = False
 
     # Including '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' in the cmake args
     # causes bits of cmake's output to end up in the autoconf-generated

--- a/var/spack/repos/builtin/packages/kallisto/package.py
+++ b/var/spack/repos/builtin/packages/kallisto/package.py
@@ -13,8 +13,29 @@ class Kallisto(CMakePackage):
     homepage = "http://pachterlab.github.io/kallisto"
     url      = "https://github.com/pachterlab/kallisto/archive/v0.43.1.tar.gz"
 
+    version('0.46.2', sha256='c447ca8ddc40fcbd7d877d7c868bc8b72807aa8823a8a8d659e19bdd515baaf2')
     version('0.43.1', sha256='7baef1b3b67bcf81dc7c604db2ef30f5520b48d532bf28ec26331cb60ce69400')
 
     depends_on('zlib')
     depends_on('hdf5')
     depends_on('mpich')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    # Including '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' in the cmake args
+    # causes bits of cmake's output to end up in the autoconf-generated
+    # configure script.
+    # See https://github.com/spack/spack/issues/15274 and
+    # https://github.com/pachterlab/kallisto/issues/253
+    @property
+    def std_cmake_args(self):
+        """Call the original std_cmake_args and then filter the verbose
+        setting.
+        """
+        a = super(Kallisto, self).std_cmake_args
+        args = [i for i in a if i != '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON']
+
+        return args

